shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/distsql/rdl/resource/DropResourceBackendHandlerTest.java

```diff
@@ -0,0 +1,100 @@
+import org.apache.shardingsphere.infra.context.metadata.MetaDataContexts;
+import org.apache.shardingsphere.infra.distsql.exception.DistSQLException;
+import org.apache.shardingsphere.infra.metadata.ShardingSphereMetaData;
+import org.apache.shardingsphere.infra.metadata.resource.ShardingSphereResource;
+import org.apache.shardingsphere.infra.metadata.rule.ShardingSphereRuleMetaData;
+import org.apache.shardingsphere.proxy.backend.communication.jdbc.connection.BackendConnection;
+import org.apache.shardingsphere.proxy.backend.context.ProxyContext;
+import org.apache.shardingsphere.proxy.backend.response.header.ResponseHeader;
+import org.apache.shardingsphere.proxy.backend.response.header.update.UpdateResponseHeader;
+import org.apache.shardingsphere.transaction.context.TransactionContexts;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.sql.DataSource;
+import java.util.Collections;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DropResourceBackendHandlerTest {
+
+    @Mock
+    private DropResourceStatement dropResourceStatement;
+
+    @Mock
+    private BackendConnection backendConnection;
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private MetaDataContexts metaDataContexts;
+
+    @Mock
+    private TransactionContexts transactionContexts;
+
+    @Mock
+    private ShardingSphereMetaData metaData;
+
+    @Mock
+    private ShardingSphereResource resource;
+
+    @Mock
+    private DataSource dataSource;
+
+    @Mock
+    private ShardingSphereRuleMetaData ruleMetaData;
+
+    private DropResourceBackendHandler dropResourceBackendHandler;
+
+    @Before
+    public void setUp() throws Exception {
+        dropResourceBackendHandler = new DropResourceBackendHandler(dropResourceStatement, backendConnection);
+    }
+
+    @Test
+    public void assertExecute() throws DistSQLException {
+        ProxyContext.getInstance().init(metaDataContexts, transactionContexts);
+        when(metaDataContexts.getAllSchemaNames()).thenReturn(Collections.singleton("test"));
+        when(metaDataContexts.getMetaData("test")).thenReturn(metaData);
+        when(metaData.getResource()).thenReturn(resource);
+        when(resource.getDataSources()).thenReturn(new HashMap<String, DataSource>() {
+                {
+                    put("test0", dataSource);
+                }
+            }
+        );
+        when(metaData.getRuleMetaData()).thenReturn(ruleMetaData);
+        when(ruleMetaData.getRules()).thenReturn(Collections.emptyList());
+        ResponseHeader responseHeader = dropResourceBackendHandler.execute("test", createDropResourceStatement());
+        assertTrue(responseHeader instanceof UpdateResponseHeader);
```

Comment:
Please assert whether resources are deleted, and need to add more case for abnormal process.